rock_paper_scissors/exc.py

```diff
@@ -0,0 +1,49 @@
+class ProgramError(Exception):
+    """Base exception class for errors raised by rock paper scissors program."""
+    pass
+
+
+class UserDefinedError(ProgramError):
+    """Raised for all user input errors."""
+    pass
+
+
+class UserChoiceInputError(UserDefinedError):
+    """Raised for invalid User sting input."""
+    def __init__(self, message):
+        self.message = message
+
+
+class UserRoundsInputError(UserDefinedError):
+    """Raised for invalid rounds input."""
+    pass
+
+
+class RoundsStringError(UserRoundsInputError):
```

Comment:
I've gone back and forth a bit on this today... ultimately, I think the subclasses of the `UserRoundsInputError` exception type are unnecessary. These would be really good test cases (and ultimately, they should be), but I'm not sure that the granularity provides much value and it can contribute to "sprawl" in code when a codebase gets larger.

There may be some business cases in programming where having very granular error classes are very useful, if not outright necessary, but I think that what `RoundsStringError`, `RoundsEvenError`, `RoundsNegativeError` and `RounderGreaterThan100Error` are all providing can be achieved by simply reusing their common base class and passing a string when raising the exception that describes the problem.

```
try:
    value = int(value)
except ValueError:
    raise InvalidRoundsError('Rounds must be an integer')

try:
    assert value % 2 != 0:
except AssertionError:
    raise InvalidRoundsError('Rounds must be an odd number')

# etc...
```